ADR/0027-availability-probe-framework.md

```diff
@@ -0,0 +1,68 @@
+# 27. Availability Probe Framework
+
+* Date 2023-07-06
+
+## Status
+
+Accepted
+
+## Context
+
+As a product team, we want a composite system availability metric for both Production
+and Staging environments, so that we can make roadmap decisions with easy-to-understand
+feedback about the reliability of the system in mind.
+  
+This metric should tell us the overall up time of RHTAP with respect to services under
+the control of Red Hat.
+
+RHTAP is defined to be available at a given moment if all of its components are
+reporting to be available at that given moment.
+
+A component is defined to be available at a given moment if all of its availability
+probes are reporting to be available at that given moment.
+
+A convention is required for providing the availability of a probe.
+
+Once this is in place, those metrics can be aggregated in order to report the overall
+availability of RHTAP.
+
+## Decision
+
+Probes' availability will be provided as a Prometheus metric. The metric will contain
+labels to allow differentiating between the different probes. When exported out of the
+cluster of origin, additional labels will be attached to the metric to mark the cluster
+in which the metric was generated.
+
+### Delivery Method
+
+It is up for each team to define what it means for its component(s) to be available.
+
+Each team will define CronJobs that will test the availability of their components.
+The Job started by each CronJob will terminate successfully if the test complete
+successfully, and will terminate with an error in case the test fails.
```

Comment:
So this means we don't get anything until all teams manage to write some probes? Perhaps we need to define a fall-back to existing passive data such as the Prometehus "up" metrics?

Reply:
I suppose we can do something along those lines for the time being, but I don't think it should go inside the ADR.